vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,599 @@
+The netCDF will be created with an automatically generated *site* dimension,
+and lat and long variables affiliated with each site, similar to [this example](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#_orthogonal_multidimensional_array_representation_of_time_series).
+
+The crs_attribute list provides more information about the CRS.
+
+## Dimension Data
+
+NetCDFs are organized into dimensions. Dimensions have two key components: Bounds
+and values. The dimensions also have other information which is designated in the
+*Attribute Data* (see below).
+
+The bounds and values arguments are critical metadata that need to match the
+data in the array. Bounds data are always written as a contiguous vector,
+represent the min and max of the bounds, and should overlap with one another.
+The expected organization is c(start1, end1, start2, end2, ..., startn, endn)
+
+For example, if you have an array that contains time series data
+(n = 10 measurements) of one variable, then the values in the time attribute data
+needs to be of length 10 the bounds needs to be of length 20.
+
+See http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#cell-boundaries
+for more information.
+
+### Latitude (Y) and Longitude (X) Dimensions
+
+NetCDFs created with *gridded* locations will have at least two dimensions: latitude and longitude.
+
+The bounds and values of the latitude and longitude are determined from either the grid
+or the location data within the *create_empty_netCDF_file* function.
+
+### Site (i) Dimension
+
+NetCDFs created *without* gridded locations will have at least one dimensions: site.
+
+Site numbers will be automatically assigned within the *create_empty_netCDF_file* function,
+and have affiliated lat and long variables.
+
+### Time (t) Dimension
+
+The user must define the bounds and values for the time dimension, as inputs to the
+*create_empty_netCDF_file*.
+
+The bounds of the time dimension are the start and end of each interval of the time dimension.
+
+The values refer to the midpoint of each interval of the time dimension.
+
+The bounds and values need to be in the same units.
+
+A common unit for netCDF is days since 1900. A full list of accepted time units can
+be found [here](https://ncics.org/portfolio/other-resources/udunits2/).
+
+```{r, timeD}
+## time vals
+time_midpoints <- c(43920, 44285, 44650, 45015, 45381, 45746, 46111,
+                    46476, 46842, 47207)# mid point of year n days since 1900-01-01
+
+## time bounds
+annual_time_bounds <- c(43737, 44102, 44102, 44467, 44467, 44832, 44833, 45197,
+                        45197, 45563, 45563, 45928, 45928, 46293, 46293, 46658,
+                        46658, 47024, 47024, 47389) # beginning and end of year in days since 1900-01-01
+```
+
+### Depth (z) Dimension
+
+The user must define the bounds and values for the depth dimension.
+
+For depth values, we use the maximum depth value.
```

Comment:
line 146: Why are we using the lower soil layer depths as values for the vertical dimension? Depending on the SOILWAT2 variable, the SOILWAT2 output may refer to the entire layer width, e.g., VWC (in which case the values should reflect the layer midpoints), or to a specific depth, e.g., soil temperature (in which case we shouldn't have bounds...).